pkg/agent/monitortool/monitor.go

```diff
@@ -0,0 +1,415 @@
+		AddFunc:    m.onNodeLatencyMonitorAdd,
+		UpdateFunc: m.onNodeLatencyMonitorUpdate,
+		DeleteFunc: m.onNodeLatencyMonitorDelete,
+	})
+
+	return m
+}
+
+// onNodeLatencyMonitorAdd is the event handler for adding NodeLatencyMonitor.
+func (m *MonitorTool) onNodeLatencyMonitorAdd(obj interface{}) {
+	nlm := obj.(*v1alpha2.NodeLatencyMonitor)
+	klog.InfoS("NodeLatencyMonitor added", "NodeLatencyMonitor", klog.KObj(nlm))
+
+	// Parse the ping interval and timeout
+	pingInterval, err := time.ParseDuration(nlm.Spec.PingInterval)
+	if err != nil {
+		klog.ErrorS(err, "Failed to parse ping interval")
+		return
+	}
+	pingTimeout, err := time.ParseDuration(nlm.Spec.PingTimeout)
+	if err != nil {
+		klog.ErrorS(err, "Failed to parse ping timeout")
+		return
+	}
+
+	// Update the latency config
+	m.latencyConfig = &LatencyConfig{
+		Enable:   true,
+		Interval: pingInterval,
+		Timeout:  pingTimeout,
+		Limit:    nlm.Spec.PingConcurrentLimit,
+	}
+
+	// Notify the latency config changed
+	m.latencyConfigChanged <- struct{}{}
+}
+
+// onNodeLatencyMonitorUpdate is the event handler for updating NodeLatencyMonitor.
+func (m *MonitorTool) onNodeLatencyMonitorUpdate(oldObj, newObj interface{}) {
+	oldNLM := oldObj.(*v1alpha2.NodeLatencyMonitor)
+	newNLM := newObj.(*v1alpha2.NodeLatencyMonitor)
+	klog.InfoS("NodeLatencyMonitor updated", "NodeLatencyMonitor", klog.KObj(newNLM))
+
+	if oldNLM.GetGeneration() == newNLM.GetGeneration() {
+		return
+	}
+
+	// Parse the ping interval and timeout
+	pingInterval, err := time.ParseDuration(newNLM.Spec.PingInterval)
+	if err != nil {
+		klog.ErrorS(err, "Failed to parse ping interval")
+		return
+	}
+	pingTimeout, err := time.ParseDuration(newNLM.Spec.PingTimeout)
+	if err != nil {
+		klog.ErrorS(err, "Failed to parse ping timeout")
+		return
+	}
+
+	// Update the latency config
+	m.latencyConfig = &LatencyConfig{
+		Enable:   true,
+		Interval: pingInterval,
+		Timeout:  pingTimeout,
+		Limit:    newNLM.Spec.PingConcurrentLimit,
+	}
+
+	// Notify the latency config changed
+	m.latencyConfigChanged <- struct{}{}
+}
+
+// onNodeLatencyMonitorDelete is the event handler for deleting NodeLatencyMonitor.
+func (m *MonitorTool) onNodeLatencyMonitorDelete(obj interface{}) {
+	klog.InfoS("NodeLatencyMonitor deleted", "NodeLatencyMonitor")
+
+	// Update the latency config
+	m.latencyConfig = &LatencyConfig{Enable: false}
+
+	// Notify the latency config changed
+	m.latencyConfigChanged <- struct{}{}
+}
+
+func getRate(count int, d time.Duration, burst int) *rate.Limiter {
+	r := float64(count) / (d.Seconds() * 0.9)
+	return rate.NewLimiter(rate.Limit(r), burst)
+}
+
+func (m *MonitorTool) pingAll() {
+	// Get all node internal/external IP.
+	nodeIPs := m.latencyStore.ListNodeIPs()
+
+	// TODO: Get current node internal/external IP.
+	var fromIP string
+	if m.gatewayConfig.IPv4 != nil {
+		fromIP = m.gatewayConfig.IPv4.String()
+	} else {
+		fromIP = m.gatewayConfig.IPv6.String()
```

Comment:
Again this should not a if/else condition

Reply:
Ok